includes/date-functions.php

```diff
@@ -80,6 +55,48 @@ function edd_get_timezone() {
  *
  * @return string date_format()-compatible date format string.
  */
-function edd_get_date_format( $format ) {
+function edd_get_date_format( $format = 'date' ) {
 	return EDD()->utils->get_date_format_string( $format );
 }
+
+/**
+ * Get the format used by jQuery UI Datepickers.
+ *
+ * Use this if you need to use placeholder or format attributes in input fields.
+ *
+ * This is a bit different than `edd_get_date_format()` because these formats
+ * are exposed to users as hints and also used by jQuery UI so the Datepicker
+ * knows what format it returns into it's connected input value.
+ *
+ * Previous to this function existing, all values were hard-coded, causing some
+ * inconsistencies across admin-area screens.
+ *
+ * @see https://github.com/easydigitaldownloads/easy-digital-downloads/commit/e9855762892b6eec578b0a402f7950f22bd19632
+ *
+ * @since 3.0
+ *
+ * @param string $context The context we are getting the format for. Accepts 'display' or 'js'.
+ *                        Use 'js' for use with jQuery UI Datepicker. Use 'display' for HTML attributes.
+ * @return string
+ */
+function edd_get_date_picker_format( $context = 'display' ) {
+
+	// What is the context that we are getting the picker format for?
+	switch ( $context ) {
+
+		// jQuery UI Datepicker does its own thing
+		case 'js' :
+		case 'javascript' :
+			$retval = EDD()->utils->get_date_format_string( 'date-js' );
```

Comment:
Same question on `$retval` vs something self-documenting like `$format`.

Reply:
Because it’s a return value based on the context.

Naming it something like `$retval` informs developers that that’s the variable that will inevitably be returned by the function.

Naming it `$format` does not indicate its the return value. It also possibly collides with other local variables or parameters.

`$retval` is like a reserved keyword, IMO. Use it only to define the variable used to filter and return something.